adapters/zemanta/zemanta.go

```diff
@@ -0,0 +1,147 @@
+package zemanta
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Zemanta adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	reqCopy := *request
+
+	var errs []error
+	var zemantaExt openrtb_ext.ExtImpZemanta
+	for i := 0; i < len(reqCopy.Imp); i++ {
+		imp := reqCopy.Imp[i]
+
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if err := json.Unmarshal(bidderExt.Bidder, &zemantaExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		imp.TagID = zemantaExt.TagId
+		reqCopy.Imp[i] = imp
+	}
+
+	publisher := &openrtb.Publisher{
+		ID:     zemantaExt.Publisher.Id,
+		Name:   zemantaExt.Publisher.Name,
+		Domain: zemantaExt.Publisher.Domain,
+	}
```

Comment:
Why does this adapter need to define it's own publisher object, instead of using what's provided on the request. Same question for bcat and badv.

Reply:
This is mostly regarding support for s2s in Prebid.js. We already support the publisher param in our Prebid.js and are now adding support for bcat and badv in https://github.com/prebid/Prebid.js/pull/6468. Is there any other way to support these params from Prebid.js? AFAIK they come from the bidder ext field in the request.

Reply:
I'm not familiar with the details of the Prebid.js interaction in this case. @AlexBVolcy is investigating. Does Prebid.js not allow publishers to specify request level `badv` and `bcat`?

Reply:
So after some investigating, it seems that this work around that you have implemented is necessary for the time being! There's nothing currently in the works with Prebid.js that allows for a more general way of supporting `badv` and `bcat`. Hopefully in the future this changes!